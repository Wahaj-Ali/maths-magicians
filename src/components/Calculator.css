.calPage {
  display: flex;
  justify-content: center;
  padding: 15px;
}

.calculator {
  display: flex;
  max-width: 320px;
  align-self: center;
  flex-wrap: nowrap;
  padding: 0 90px;
}

.calculator__display {
  padding: 10px;
  font-size: 1.3em;
  text-align: right;
  background-color: grey;
  overflow: hidden;
}

.calculatorUI {
  display: flex;
  flex-wrap: wrap;
}

button {
  flex: 0 1 25%;
  height: 70px;
  color: #111;
  font-size: 2em;
  background: #e0e0e0;
  border: 1px solid #c7c7c7;
}

button:hover {
  border: 2px solid #b42e10;
}

.btn--opaque {
  background-color: #f5913e;
}

.btn--large {
  flex: 0 1 50%;
}

@media only screen and (max-width: 768px) {
  .calculator {
    padding: 0;
  }
}
