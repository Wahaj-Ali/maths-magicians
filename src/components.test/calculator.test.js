import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

describe('Calculator', () => {
  it('renders Calculator component', () => {
    render(<Calculator />);
  });
});

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('checks UI changes state after the button is clicked', () => {
  const { getByText } = render(<Calculator />);
  const number2 = getByText(2);
  const answer = number2.textContent;
  expect(answer).toBe('2');
});
