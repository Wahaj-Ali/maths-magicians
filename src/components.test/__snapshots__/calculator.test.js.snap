// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="calPage"
>
  <div
    className="calculator"
  >
    <div
      className="calculator__wrap"
    >
      <div
        className="calculator__display"
      >
        0
      </div>
      <div
        className="calculatorUI"
      >
        <button
          onClick={[Function]}
          type="button"
          value="clear"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="pos-neg"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="btn--opaque"
          onClick={[Function]}
          type="button"
          value="/"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="btn--opaque"
          onClick={[Function]}
          type="button"
          value="x"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="btn--opaque"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="btn--opaque"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
        <button
          className="btn--large"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
        <button
          className="btn--opaque"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
